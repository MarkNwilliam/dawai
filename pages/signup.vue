<template>
  <v-container fluid grid-list-xs>
    <v-layout fill-height row wrap justify-center align-center>
      <v-flex xs12>
        <v-card class="mx-auto" :loading="loading" max-width="500">
          <v-card-title class="title font-weight-regular justify-space-between">
            <span>{{ currentTitle }}</span>
            <v-avatar
              color="primary lighten-2"
              class="subheading white--text"
              size="28"
              v-text="step"
            ></v-avatar>
          </v-card-title>

          <v-window v-model="step">
            <v-window-item :value="1">
              <v-card-text>
                <v-text-field
                  v-model="email"
                  :clear-icon="mdiClose"
                  clearable
                  outlined
                  label="Email"
                  :prepend-inner-icon="mdiEmail"
                  hint="This is the email you will use to login to your Dawai account"
                  :persistent-hint="false"
                ></v-text-field>
                <span class="caption grey--text text--darken-1">
                  By clicking next you agree to our
                  <nuxt-link to="/terms">Terms and Conditions</nuxt-link>.
                </span>
              </v-card-text>
            </v-window-item>

            <v-window-item :value="2">
              <v-card-text>
                <v-text-field
                  v-model="password"
                  label="Password"
                  :prepend-inner-icon="mdiShieldLock"
                  :clear-icon="mdiClose"
                  clearable
                  outlined
                  type="password"
                ></v-text-field>
                <v-text-field
                  v-model="secondPassword"
                  label="Confirm Password"
                  :prepend-inner-icon="mdiShieldLockOutline"
                  :clear-icon="mdiClose"
                  clearable
                  outlined
                  type="password"
                ></v-text-field>
                <span class="caption grey--text text--darken-1">
                  Enter a strong, 8 minimum character password twice
                </span>
              </v-card-text>
            </v-window-item>

            <v-window-item :value="3">
              <div class="pa-4 text-center">
                <v-img
                  class="mb-4"
                  contain
                  height="200"
                  src="/logo-1024-transparent.svg"
                ></v-img>
                <h3 class="title font-weight-light mb-2">Welcome to Dawai</h3>
                <span class="caption grey--text"
                  >Thanks for signing up! We'll get back to you via E-mail for
                  account activation.</span
                >
              </div>
            </v-window-item>
          </v-window>

          <!-- <v-divider></v-divider> -->

          <v-card-actions class="px-6">
            <v-btn :disabled="step === 1" large text @click="step--">
              Back
            </v-btn>
            <div class="flex-grow-1"></div>
            <v-btn
              :disabled="step === 3"
              color="primary"
              large
              depressed
              @click="nextStep"
            >
              {{ step === 3 ? 'Finished!' : 'Next' }}
            </v-btn>
          </v-card-actions>
        </v-card>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
import {
  mdiEmail,
  mdiClose,
  mdiShieldLock,
  mdiShieldLockOutline
} from '@mdi/js'
export default {
  layout: 'none',
  head: () => ({
    title: 'Sign Up',
    meta: [
      {
        hid: 'description',
        name: 'description',
        content: 'Sign Up for using the Dawai Pharmacy Management Software'
      }
    ]
  }),
  data: () => ({
    mdiEmail,
    mdiClose,
    mdiShieldLock,
    mdiShieldLockOutline,
    step: 1,
    loading: false,
    email: '',
    password: '',
    secondPassword: ''
  }),

  computed: {
    currentTitle() {
      switch (this.step) {
        case 1:
          return 'Sign-up'
        case 2:
          return 'Create a password'
        default:
          return 'Account created!'
      }
    }
  },

  methods: {
    nextStep() {
      this.loading = true
      setTimeout(() => {
        this.step++
        this.loading = false
      }, 2000)
    }
  }
}
</script>
