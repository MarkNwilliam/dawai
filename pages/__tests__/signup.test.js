// Libraries
import Vuetify from 'vuetify'

// Component
import { mount, createLocalVue } from '@vue/test-utils'
import signup from '../signup'

// Utilities
const localVue = createLocalVue()

describe('signup.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should render', done => {
    const wrapper = mount(signup, {
      localVue,
      vuetify
    })

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot()

    done()
  })
})