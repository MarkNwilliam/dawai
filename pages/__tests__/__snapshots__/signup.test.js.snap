// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup.vue should render 1`] = `
"<v-container fluid=\\"\\" grid-list-xs=\\"\\"><v-layout fill-height=\\"\\" row=\\"\\" wrap=\\"\\" justify-center=\\"\\" align-center=\\"\\"><v-flex xs12=\\"\\"><v-card max-width=\\"500\\" class=\\"mx-auto slight-br\\"><v-card-title class=\\"title font-weight-regular justify-space-between\\"><span>Sign-up</span> <v-avatar color=\\"primary lighten-2\\" size=\\"28\\" class=\\"subheading white--text\\">1</v-avatar></v-card-title> <v-window><v-window-item value=\\"1\\"><v-card-text><v-text-field clear-icon=\\"M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z\\" clearable=\\"\\" outlined=\\"\\" rounded=\\"\\" label=\\"Email\\" prepend-inner-icon=\\"M20,8L12,13L4,8V6L12,11L20,6M20,4H4C2.89,4 2,4.89 2,6V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V6C22,4.89 21.1,4 20,4Z\\" hint=\\"This is the email you will use to login to your Dawai account\\"></v-text-field> <span class=\\"caption grey--text text--darken-1\\">
                By clicking next you agree to our
                <nuxt-link to=\\"/terms\\">Terms and Conditions</nuxt-link>.
              </span></v-card-text></v-window-item> <v-window-item value=\\"2\\"><v-card-text><v-text-field label=\\"Password\\" prepend-inner-icon=\\"M12,1L3,5V11C3,16.55 6.84,21.74 12,23C17.16,21.74 21,16.55 21,11V5L12,1M12,7C13.4,7 14.8,8.1 14.8,9.5V11C15.4,11 16,11.6 16,12.3V15.8C16,16.4 15.4,17 14.7,17H9.2C8.6,17 8,16.4 8,15.7V12.2C8,11.6 8.6,11 9.2,11V9.5C9.2,8.1 10.6,7 12,7M12,8.2C11.2,8.2 10.5,8.7 10.5,9.5V11H13.5V9.5C13.5,8.7 12.8,8.2 12,8.2Z\\" clear-icon=\\"M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z\\" clearable=\\"\\" outlined=\\"\\" rounded=\\"\\" type=\\"password\\"></v-text-field> <v-text-field label=\\"Confirm Password\\" prepend-inner-icon=\\"M21,11C21,16.55 17.16,21.74 12,23C6.84,21.74 3,16.55 3,11V5L12,1L21,5V11M12,21C15.75,20 19,15.54 19,11.22V6.3L12,3.18L5,6.3V11.22C5,15.54 8.25,20 12,21M14.8,11V9.5C14.8,8.1 13.4,7 12,7C10.6,7 9.2,8.1 9.2,9.5V11C8.6,11 8,11.6 8,12.2V15.7C8,16.4 8.6,17 9.2,17H14.7C15.4,17 16,16.4 16,15.8V12.3C16,11.6 15.4,11 14.8,11M13.5,11H10.5V9.5C10.5,8.7 11.2,8.2 12,8.2C12.8,8.2 13.5,8.7 13.5,9.5V11Z\\" clear-icon=\\"M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z\\" clearable=\\"\\" outlined=\\"\\" rounded=\\"\\" type=\\"password\\"></v-text-field> <span class=\\"caption grey--text text--darken-1\\">
                Enter a strong, 8 minimum character password twice
              </span></v-card-text></v-window-item> <v-window-item value=\\"3\\"><div class=\\"pa-4 text-center\\"><v-img contain=\\"\\" height=\\"200\\" src=\\"/logo-1024-transparent.svg\\" class=\\"mb-4\\"></v-img> <h3 class=\\"title font-weight-light mb-2\\">Welcome to Dawai</h3> <span class=\\"caption grey--text\\">Thanks for signing up! We'll get back to you via E-mail for
                account activation.</span></div></v-window-item></v-window> <v-card-actions class=\\"px-6\\"><v-btn rounded=\\"\\" disabled=\\"true\\" large=\\"\\" text=\\"\\">
            Back
          </v-btn> <div class=\\"flex-grow-1\\"></div> <v-btn color=\\"primary\\" large=\\"\\" depressed=\\"\\" rounded=\\"\\">
            Next
          </v-btn></v-card-actions></v-card></v-flex></v-layout></v-container>"
`;
